src/main/resources/com/rultor/spi/talk.xsd

```diff
@@ -30,43 +30,113 @@
+                </xs:annotation>
+            </xs:element>
             <xs:element name="dir" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation source="description">This element define the directory of the daemon.</xs:documentation>
+                </xs:annotation>
                 <xs:simpleType>
                     <xs:restriction base="xs:string">
                         <xs:pattern value="[0-9a-zA-Z_\\:/+\.-]+"/>
                     </xs:restriction>
                 </xs:simpleType>
             </xs:element>
-            <xs:element name="ended" type="xs:dateTime" minOccurs="0"/>
-            <xs:element name="code" type="xs:integer" minOccurs="0"/>
-            <xs:element name="highlights" type="xs:string" minOccurs="0"/>
+            <xs:element name="ended" type="xs:dateTime" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the ended time of the daemon.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="code" type="xs:integer" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the code of the daemon.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="highlights" type="xs:string" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the description of the daemon.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
         </xs:all>
         <xs:attribute name="id" type="hash" use="required"/>
     </xs:complexType>
     <xs:complexType name="shell">
+        <xs:annotation>
+            <xs:documentation source="description">This type describe the specification about the shell task.</xs:documentation>
+        </xs:annotation>
         <xs:all>
-            <xs:element name="host" type="xs:string"/>
-            <xs:element name="port" type="xs:integer"/>
-            <xs:element name="login" type="xs:string"/>
-            <xs:element name="key" type="xs:string"/>
+            <xs:element name="host" type="xs:string">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the host name or IP address.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="port" type="xs:integer">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the port to use in the connection.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="login" type="xs:string">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the user to use in the connection.</xs:documentation>
```

Comment:
I think it's better to restate it as "This element has the username of the connection"